### ISSUE TITLE
how to use Cleanlab with GPU

### ISSUE THREAD

**Issue by nanfei666** [cleanlab/cleanlab#555]
hi, i can't find a way to use pytorch with GPU in Cleanlab to accelerate the process. it takes days to find label issue when i use cross_val_predict as this
```python
train_dataset - datasets.ImgaeFolder(traindir,transforms.compose(...))
x=[]
for images,target in train_dataset:
   x.append(images.numpy())
y = np.array(train_dataset.targets).astype("int64")

model_skorch = model.__dict__['resnet50'](pretrained=False,num_classes=4)
net =NeuralNetClassifier(
model_skorch,
max_epochs=300,
criterion=nn.VrossEntropyLoss(),
lr=0.01,
iterator_train_shuffle=True,)
pred_probs = cross_val_predict(net,X,y,cv=3,method='predict_proba')
```

could you please teach me how to speed up this process, thank you

**Comment by cgnorthcutt** [comment]
Hi @nanfei666 how big is your dataset? How long does it take to train without using cross validation?

**Comment by nanfei666** [comment]
> Hi @nanfei666 how big is your dataset? How long does it take to train without using cross validation?

i've got around 87000 images in total, and it takes 2 days to finish one training process when i use cross validation. while training with pytorch it only takes hours with multi-gpu

**Comment by jwmueller** [comment]
@nanfei666 One option is to just run the cross-validation yourself outside of cleanlab to get out-of-sample predicted class probabilities for every image. Then you can just feed these into cleanlab's `find_label_issues()` function which runs very quickly (this function does not require GPU). 

See these links for examples of this:
https://docs.cleanlab.ai/stable/tutorials/pred_probs_cross_val.html
https://docs.cleanlab.ai/stable/tutorials/image.html

**Comment by nanfei666** [comment]
@jwmueller much appreciate,it's much faster after i train it based on the methods you provided. however after i get the `pred_probs` and try to find label issue using CleanLab api `find_label_issue()`, it takes a long time to produce results. the images number is 87433. Is there any way to speed up this step? thanks a lot

**Comment by cgnorthcutt** [comment]
Hmm @nanfei666 `cleanlab.filter.find_label_issues()` should be very fast I'd you provide `pred_probs`. How many classes do you have?

**Comment by nanfei666** [comment]
thanks for your help , the problem has been solved, close for now

**Comment by Jordan-Pierce** [comment]
@nanfei666 Would you mind sharing how you solved the problem for those with the same question?
